plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/query/DynamicTable.java

```diff
@@ -28,23 +28,32 @@
 public final class DynamicTable
 {
     private final String tableName;
+
     private final Optional<String> suffix;
+
     private final List<String> selections;
+
+    private final Optional<String> filter;
+
+    // semantically aggregation is applied after constraint
     private final List<String> groupingColumns;
     private final List<AggregationExpression> aggregateColumns;
+
     private final List<OrderByExpression> orderBy;
+
+    // semantically limit is applied after aggregation
     private final OptionalLong limit;
     private final OptionalLong offset;
-    private final Optional<String> filter;
+
     private final String query;
```

Comment:
should `query` be moved before filter, aggregations, sorting and limit?

ie are filter, aggregations, sorting and limit applied on top of a query?

Reply:
what about `selections`?

Reply:
The query, if present, is the "passthrough" query - which is then parsed by pinot. The result will only contain `selections` if there are no aggregations. 
If there are pushdowns, then the query will not be present.

Reply:
A little more context: for passthrough queries, DynamicTableBuilder takes the query, parses it with the pinot broker request parser and the other fields in DynamicTable (selections, filter, grouping columns, limit) are created based on the parsed query. 
If the DynamicTable was created via pushdown then the query will be empty and the other fields will be based on the result of the pushdown.